server/rest/src/main/java/org/infinispan/rest/Http11To2UpgradeHandler.java

```diff
@@ -79,9 +82,11 @@ private void configureHttp2(ChannelPipeline pipeline) {
    }
 
    private void configureHttp1(ChannelPipeline pipeline) {
-      final HttpServerCodec httpCodec = new HttpServerCodec(MAX_INITIAL_LINE_SIZE, MAX_HEADER_SIZE, restServer.getConfiguration().maxContentLength());
+      RestServerConfiguration configuration = restServer.getConfiguration();
+      final HttpServerCodec httpCodec = new HttpServerCodec(MAX_INITIAL_LINE_SIZE, MAX_HEADER_SIZE, configuration.maxContentLength());
       pipeline.addLast(httpCodec);
       pipeline.addLast(new CorsHandler(corsConfig));
+      pipeline.addLast(new HttpContentCompressor(configuration.getCompressionLevel()));
```

Comment:
Should we only optionally add this if the level is > 0?

Reply:
In theory level 0 is a valid level of compression, so I think level 0 != no compression. Wonder if level 0 will actually make the payload larger, since it will not compress it but will add gzip/lzip headers :)